### PR TITLE
Add Azure/VXLAN support

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/16283

### CODE REVIEW

**Comment by kubasobon** [comment]
```
~ k --kubeconfig=kubacapz-kubeconfig get pods -A -lapp=calico
NAMESPACE     NAME                                       READY   STATUS             RESTARTS   AGE
kube-system   calico-kube-controllers-8658d75748-d88dl   1/1     Running            0          16m
kube-system   calico-node-dbzz2                          0/1     CrashLoopBackOff   7          16m
kube-system   calico-node-gpcsr                          0/1     CrashLoopBackOff   7          16m
kube-system   calico-node-r8v2q                          0/1     CrashLoopBackOff   7          16m
```

**Comment by kubasobon** [comment]
Fixed :wrench::
```
k --kubeconfig=kubacapz-kubeconfig get pods -n kube-system -lapp=calico
NAME                                       READY   STATUS    RESTARTS   AGE
calico-kube-controllers-6c45b87cfb-pjdvk   1/1     Running   0          2m52s
calico-node-cbqfz                          1/1     Running   0          2m57s
calico-node-nskhn                          1/1     Running   0          2m57s
calico-node-xzhdt                          1/1     Running   0          2m57s
```

File: helm/calico-app/templates/rbac.yaml [hunk 1]
```diff
@@ -100,6 +100,15 @@ rules:
       - namespaces
     verbs:
       - get
+  {{- if eq .Values.provider "azure" }}
```
Comment (JosephSalisbury) [inline helm/calico-app/templates/rbac.yaml:103]: For my curiosity, why do we need this?

**Review by JosephSalisbury — APPROVED** [review]

File: helm/calico-app/templates/rbac.yaml (hunk 1 above)
Comment (kubasobon) [reply helm/calico-app/templates/rbac.yaml:103]: `# Needed when configuring bgp password in bgppeer`
Source: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/templates/addons/calico.yaml#L3389

Comment (JosephSalisbury) [reply helm/calico-app/templates/rbac.yaml:103]: It might be helpful to retain that comment

Comment (kubasobon) [reply helm/calico-app/templates/rbac.yaml:103]: Done

**Pull request merged**